### PR TITLE
add a Dockerfile

### PR DESCRIPTION
**Pull request by JensTimmerman**
This file contains all the dependencies for running the tests I have to install on a minimal centos 7 install image.
It's usefull to run tests on a non centos machine.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+# Set the working directory to install dependencies to /quattor
+WORKDIR /quattor
+
+# install library core in /quattor, tests need it
+ADD https://codeload.github.com/quattor/template-library-core/tar.gz/master /quattor/template-library-core-master.tar.gz
+RUN tar xvfz template-library-core-master.tar.gz
+
+# Copy the current directory contents into the container at /ncm-metaconfig
+#TODO: use docker volumes here to mount the current directory contents, no no need to rebuild every time to test
+ADD . /ncm-metaconfig/
```
Comment (stdweird) [inline Dockerfile]: this needs more comment why this works (ie you are expected to run this from the ncm-metaconfig dir of the clone you are developing in)

Comment (JensTimmerman) [reply Dockerfile]: indeed, I'll mark this as WIP until I fix the todo and document some more

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+# Install dependencies
+RUN yum groupinstall 'Development Tools' -y
```
Comment (stdweird) [inline Dockerfile]: is this needed? all deps should be installed via other packages

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+RUN yum-config-manager --add-repo http://yum.quattor.org/current
```
Comment (stdweird) [inline Dockerfile]: this and next line are not needed if you use the `quattor-release` rpm

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+RUN yum-config-manager --add-repo http://yum.quattor.org/externals/noarch/el7/
+RUN yum install maven wget epel-release -y
+
+RUN wget http://yum.quattor.org/devel/quattor-release-1-1.noarch.rpm
```
Comment (stdweird) [inline Dockerfile]: you can do `rpm -U http://yum.quattor.org/devel/quattor-release-1-1.noarch.rpm`, no need to wget

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+# Copy the current directory contents into the container at /ncm-metaconfig
+#TODO: use docker volumes here to mount the current directory contents, no no need to rebuild every time to test
+ADD . /ncm-metaconfig/
+
+# Install dependencies
+RUN yum groupinstall 'Development Tools' -y
+RUN yum-config-manager --add-repo http://yum.quattor.org/current
+RUN yum-config-manager --add-repo http://yum.quattor.org/externals/noarch/el7/
+RUN yum install maven wget epel-release -y
```
Comment (stdweird) [inline Dockerfile]: is wget needed?

File: Dockerfile [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+RUN wget http://yum.quattor.org/devel/quattor-release-1-1.noarch.rpm
+RUN yum localinstall --nogpgcheck quattor-release-1-1.noarch.rpm -y
+RUN yum install --nogpgcheck panc ncm-ncd perl-Test-Quattor -y
```
Comment (stdweird) [inline Dockerfile]: `ncm-ncd` is very likely a dep of the 2nd rpm.
also the panc is signed, only perl-Test-Quattor not (yet)

File: Dockerfile [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+# needed by some tests, not a dependency of perl-Test-Quattor
+RUN yum install perl-JSON-Any -y
```
Comment (stdweird) [inline Dockerfile]: i can add it :wink:

File: Dockerfile [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+
+# these are not by default in centos7, but quattor tests assume they are
```
Comment (stdweird) [inline Dockerfile]: can you open an issue in CAF about this. it's a bug, should be handled cleanly

File: Dockerfile [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+RUN touch /usr/sbin/selinuxenabled /sbin/restorecon
+RUN chmod +x /usr/sbin/selinuxenabled /sbin/restorecon
+
+# Define environment variable
+ENV QUATTOR_TEST_SUITE_FILTER simple
```
Comment (stdweird) [inline Dockerfile]: this limits the tests to test profiles named `simple`. it's only needed when you ant to test a single profile that is part of a service

**Comment by stdweird** [comment]
@jrha this relies on a `quattor-release` rpm with current release repo, that i will open a PR for.

**Comment by jrha** [comment]
As discussed at workshop: "rpm with current release repo" means an RPM which points at the latest yum repository (i.e. http://yum.quattor.org/current/)

**Pull request merged**